src/item.cpp

```diff
@@ -197,25 +197,15 @@ item::item(JsonObject &jo)
     deserialize(jo);
 }
 
-item::~item()
-{
-}
-
-void item::make( const std::string new_type, bool scrub )
+item& item::convert( const itype_id& new_type )
 {
     type = find_type( new_type );
-    contents.clear();
-
-    if (scrub) {
-        components.clear();
-        charges = -1;
-        bday = 0;
-        name = "";
-        curammo = NULL;
+    if( is_null() ) {
```

Comment:
I'm pretty sure what Coolthulu means is checked exceptions, which is
something that makes exceptions in Java tolerable. Otherwise it's a
nightmare because you never know if an exception is going to forcibly exit
you from a function part way through because some sub-method threw an
exception.
We never want to bail out of the whole game, if an item is undefined we can
make up a fake item and keep going, but make enough noise that it will be
noticed immediately.
Because of this, very nearly every part of the game would be either
throwing or catching exceptions, and they would only be propagating a frame
or two.
I don't understand how exceptions violate encapsulation less than null
objects, it's reasonable to know the api of things you xall, but exceptions
propagate the api of a method potentially all the way up the stack, which
is incredibly hostile to encapsulation, unless you always catch and
re-throw exceptions at every level, at which point you might as well just
propagate errors locally.
Exceptions might make code cleaner locally, but IMO they make it harder to
understand globally, which is much worse.